contracts/Joint.sol

```diff
@@ -182,54 +187,190 @@ contract Joint {
+
+        // If returns are equal, nothing to sell
+        if (percentReturnA == percentReturnB) return (address(0), 0);
+
+        emit PnL(percentReturnA, percentReturnB);
+
+        (uint256 _AForB, uint256 _BForA) = getSpotExchangeRates();
+
+        uint256 numerator;
+        uint256 denominator;
+        if (percentReturnA > percentReturnB) {
+            _sellToken = tokenA;
+            numerator = currentA.sub(startingA.mul(currentB).div(startingB));
+            denominator = 1e18 + startingA.mul(_BForA).div(startingB);
         } else {
-            distributeProfit();
+            _sellToken = tokenB;
+            numerator = currentB.sub(startingB.mul(currentA).div(startingA));
+            denominator = 1e18 + startingB.mul(_AForB).div(startingA);
         }
+        _sellAmount = numerator.mul(1e18).div(denominator);
+        emit SellToBalance(_sellToken, _sellAmount);
     }
 
-    function createLP() internal {
-        IUniswapV2Router02(router).addLiquidity(
-            tokenA,
-            tokenB,
-            balanceOfA(),
-            balanceOfB(),
-            0,
-            0,
-            address(this),
-            now
-        );
+    function getPnL(
+        uint256 currentA,
+        uint256 currentB,
+        uint256 startingA,
+        uint256 startingB
+    ) internal pure returns (uint256 _a, uint256 _b) {
+        _a = currentA.mul(1e4).div(startingA);
+        _b = currentB.mul(1e4).div(startingB);
+    }
+
+    function getSpotExchangeRates()
+        public
+        view
+        returns (uint256 _AForB, uint256 _BForA)
+    {
+        (uint256 reserve0, uint256 reserve1, ) = pair.getReserves();
+        uint256 _0For1 = reserve0.mul(1e18).div(reserve1);
+        uint256 _1For0 = reserve1.mul(1e18).div(reserve0);
+        if (pair.token0() == tokenA) {
+            _AForB = _0For1;
+            _BForA = _1For0;
+        } else {
+            _BForA = _0For1;
+            _AForB = _1For0;
+        }
     }
 
-    function setMasterChef(address _masterchef) external onlyGov {
+    function createLP()
+        internal
+        returns (
+            uint256,
+            uint256,
+            uint256
+        )
+    {
+        return
+            IUniswapV2Router02(router).addLiquidity(
+                tokenA,
+                tokenB,
+                balanceOfA(),
+                balanceOfB(),
+                0,
+                0,
+                address(this),
+                now
+            );
+    }
+
+    function setMasterChef(address _masterchef) external onlyGovernance {
         masterchef = IMasterchef(_masterchef);
         IERC20(getPair()).approve(_masterchef, type(uint256).max);
     }
 
-    function setPid(uint256 _newPid) external onlyGov {
+    function setPid(uint256 _newPid) external onlyGovernance {
         pid = _newPid;
     }
 
-    function setWETH(address _weth) external onlyGov {
+    function setWETH(address _weth) external onlyGovernance {
```

Comment:
should we remove these?
It feels like if we change any of those, we should clone a new joint.

Wdyt?

Reply:
agree, they are dead to me